### PR TITLE
feat: add C implementation for `math/base/special/csch`

### LINKED ISSUE #1816: [RFC]: add C implementation for `@stdlib/math/base/special/csch`
### Description

This RFC proposes adding C implementation for [`@stdlib/math/base/special/csch`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/csch).

### Related Issues

Related issues #649.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by gunjjoshi**
Resolves #1816 

## Description
This RFC adds C implementation for [`@stdlib/math/base/special/csch`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/csch).

> What is the purpose of this pull request?

This pull request:

-   adds C implementation for `@stdlib/math/base/special/csch`.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves a part of #649 
-   fixes #1816 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   ☑️ Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/special/csch/test/test.native.js [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+var csch = tryRequire( resolve( __dirname, './../lib/native.js' ) );
+var opts = {
+	'skip': ( csch instanceof Error )
+};
+
+
+// FIXTURES //
+
+var data = require( './fixtures/julia/data.json' );
+var largeNegative = require( './fixtures/julia/large_negative.json' );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/csch/test/test.native.js:47]: Test fixtures are currently missing from this PR, which will cause these tests to fail.

**Review by Planeshifter — CHANGES_REQUESTED** [review]

File: lib/node_modules/@stdlib/math/base/special/csch/test/test.native.js (hunk 1 above)
Comment (gunjjoshi) [reply lib/node_modules/@stdlib/math/base/special/csch/test/test.native.js:47]: They were already present there, so I haven't added them in the PR. These are the same fixtures which were being used for the `JS` implementation too. So they can be imported directly.

Or am I assuming something wrong here ?

Comment (Planeshifter) [reply lib/node_modules/@stdlib/math/base/special/csch/test/test.native.js:47]: @gunjjoshi Sorry, my mistake; I checked the wrong package for the Julia benchmarks.

**Review by Planeshifter — APPROVED** [review]
Looks all good to me; thank you!

File: lib/node_modules/@stdlib/math/base/special/csch/test/test.native.js (hunk 1 above)
Comment (gunjjoshi) [reply lib/node_modules/@stdlib/math/base/special/csch/test/test.native.js:47]: Thanks for clarifying.

**Pull request merged**